MsixCore/msixmgr/msixmgr.cpp

```diff
@@ -418,16 +418,16 @@ int main(int argc, char * argv[])
                     }
                     else
                     {
-                        if (cli.GetVHDSize() == 0)
+                        ULONGLONG vhdSize = cli.GetVHDSize();

```

Comment:
I don't think, you should create a local variable and assign the new VHD Size to it, instead, you should "Set" the the data member inside the CommandLineInterface class (create a func cli.SetVHDSize() inside CommandLineInterface). The reason is, here we only have one use case where VHDSize was being used, but suppose we had multiple places where we had to use the VHDSize then, taking the VHDSize from the data member of CommandLineInterface (i.e. cli.GetVHDSize()) would be better.

Reply:
And it would be better if you do the assigning of VHDSize at the start of the Unpack Switch case. It will help in telemetry logging as well.

Reply:
> I don't think, you should create a local variable and assign the new VHD Size to it, instead, you should "Set" the the data member inside the CommandLineInterface class (create a func cli.SetVHDSize() inside CommandLineInterface). The reason is, here we only have one use case where VHDSize was being used, but suppose we had multiple places where we had to use the VHDSize then, taking the VHDSize from the data member of CommandLineInterface (i.e. cli.GetVHDSize()) would be better.

CommandLineInterface class has data members for storing values inputted by user. If we update m_vhdSize, then we won't be able to know whether user inputted the vhdSize or not, if we want to know this information later in the code. I had another option of adding a new data member "m_defaultVhdSize" to CommandLineInterface class. But, this class is present to just store the values inputted by the user as the name suggests. Hence, I feel we shouldn't add another data member to it if that data member doesn't map to an argument expected from the user in command line.

Reply:
> And it would be better if you do the assigning of VHDSize at the start of the Unpack Switch case. It will help in telemetry logging as well.

For telemetry logging, I think we should send VHDSize as "0" or "NOT_PROVIDED", if user has not inputted vhdSize. With this we will be able to observe the behavior, how many users are leveraging this option of not providing vhdSize. Calculated vhdSize is a default calculation based on fileSize. This will not give us any meaningful info with regards to telemetry in my opinion. Could you elaborate what value are we getting by sending calculatedVhdSize to telemetry?
Also, calculating vhdSize on the basis of fileSize is only relevant for the case when Vhd(x) is getting created. If we calculate vhdSize at the start of the Unpack switch case, and the fileType to be created is CIM or we don't have to create any file and it's just simple unpack, then this calculation operation would be unnecessary.

Reply:
Agreed on above mentioned points. The current changes looks good.